### PR TITLE
Add explicit file length in form data for BLOBs

### PR DESCRIPTION
**Pull request by MHHukiewitz**

### CODE REVIEW

File: src/global.ts [hunk 1]
```diff
@@ -1 +1 @@
-export const DEFAULT_API_V2 = "https://api2.aleph.im";
+export const DEFAULT_API_V2 = "http://127.0.0.1:4024/";
```
Comment (BjrInt) [inline src/global.ts:1]: Seems like you left your test node as a default API.

File: src/messages/store/publish.ts [hunk 2]
```diff
@@ -68,7 +73,7 @@ export async function Publish({
         APIServer,
         inlineRequested,
         sync,
-        fileObject,
+        fileObject as Blob | File | undefined,
```
Comment (BjrInt) [inline src/messages/store/publish.ts:76]: This does not seem right to me, I'm not sure that the type assertion is the way to go. 
Can you provide some context as of why you do this

File: src/messages/store/publish.ts [hunk 3]
```diff
@@ -168,7 +173,7 @@ type SignAndBroadcastConfiguration = {
     sync: boolean;
 };
 
-async function sendMessage(configuration: SignAndBroadcastConfiguration, file: Buffer | Blob | File): Promise<any> {
```
Comment (BjrInt) [inline src/messages/store/publish.ts]: Why did you remove "Buffer" from the possible value? It's the default value when loading a file in memory on Node.js

**Review by BjrInt — CHANGES_REQUESTED** [review]

File: src/messages/store/publish.ts (hunk 2 above)
Comment (MHHukiewitz) [reply src/messages/store/publish.ts:76]: So above we have `RequireOnlyOne` that makes sure that at least fileHash or fileObject is given. The assertion comes out of the need that if we have a fileObject, we read its content as a Buffer and then put it into a File or Blob object, depending on what it originally was. We simply drop the possibility of a Buffer being passed, which `createAndSendStoreMessage` does not accept anymore.

File: src/messages/store/publish.ts (hunk 3 above)
Comment (MHHukiewitz) [reply src/messages/store/publish.ts]: Because raw Buffers cannot be sent in a FormData format to the server. They either have to be Blobs or Files. Everything that could be done with a buffer (computing the hash) has already been done by this point.

Comment (MHHukiewitz) [reply src/messages/store/publish.ts]: And for that, users should use `Publish` instead. But yeah, it might be a bit better constructed. Up until now, the SDK was not compatible with Node.js, because `utils.ts` was using `FileReader`

Comment (BjrInt) [reply src/messages/store/publish.ts]: The tests I have done led to the conclusion that sending Buffers was supported (that's why I could assess that Node.js worked while the browser upload was crashing). 

I had an issue with V8 Garbage Collector while trying to convert a Buffer to a Blob in Node.js. This is because it allocates a new memory slot to wrap the Buffer. I don't think that using Blob as the default format is optimal if we can already use the Buffer.

Comment (MHHukiewitz) [reply src/messages/store/publish.ts]: But as I described, I _have to_ wrap it in a Blob to be able to pass it into FormData :(

File: src/messages/store/publish.ts (hunk 2 above)
Comment (BjrInt) [reply src/messages/store/publish.ts:76]: Then maybe we should separate the method into two different functions, one for IPFS cid and the other one for binary payload. This will allow us to get rid of the big overlapping type assertions

Comment (MHHukiewitz) [reply src/messages/store/publish.ts:76]: We definitely should rework the SDK at some point, but not right now tbh. ^^'

As long as it works in the tests and in the Browser, I'm happy

**Review by BjrInt — APPROVED** [review]
Looking good (for now) :+1: 
Thanks for the submission!

**Review by amalcaraz — APPROVED** [review]
LGTM
